src/dmd/mars.d

```diff
@@ -2125,3 +2168,276 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
+            // if at least 1 "include pattern" is given, then it is assumed the
+            // user only wants to include modules that were explicitly given, which
+            // changes the default behavior from inclusion to exclusion.
+            if (includeByDefault && !matchNodes[entryIndex].isExclude)
+            {
+                //fprintf(global.stdmsg, "Matcher: found 'include pattern', switching default behavior to exclusion\n");
+                includeByDefault = false;
+            }
+        }
+
+        // Add the default 1 depth matchers
+        MatcherNode[8] defaultDepth1MatchNodes = [
+            MatcherNode(true, 1), MatcherNode(Id.std),
+            MatcherNode(true, 1), MatcherNode(Id.core),
+            MatcherNode(true, 1), MatcherNode(Id.etc),
+            MatcherNode(true, 1), MatcherNode(Id.object),
+        ];
+        {
+            auto index = findSortedIndexToAddForDepth(1);
+            matchNodes.split(index, defaultDepth1MatchNodes.length);
+            matchNodes.data[index .. index + defaultDepth1MatchNodes.length] = defaultDepth1MatchNodes[];
+        }
+    }
+}
+
+/**
+ * Pattern strings are taken directly from the -i command line option,
+ * so they can either end with the traditional terminating null, or they
+ * can end with a comma.  This function returns true if the given character
+ * matches either of these.
+ * Returns:
+ *  True if the given character is '\0' or ','.
+ */
+bool endOfPattern(char c) { return c == ',' || c == '\0'; }
+
+/**
+ * Determines the depth of the given match string.
+ * Params:
+ *  matchString = The match string to determine the depth of.
+ * Returns:
+ *  The component depth of the given match string.
+ */
+private ushort parseMatchStringDepth(const(char)* matchString)
+{
+    if (matchString[0] == '-')
+        matchString++;
+
+    // handle special case
+    if (matchString[0] == '.' && endOfPattern(matchString[1]))
+        return 0;
+
+    ushort depth = 1;
+    for (;; matchString++)
+    {
+        auto c = *matchString;
+        if (c == '.')
+            depth++;
+        if (endOfPattern(c))
+            return depth;
+    }
+}
+unittest
+{
+    assert(".".parseMatchStringDepth == 0);
+    assert("-.".parseMatchStringDepth == 0);
+    assert("abc".parseMatchStringDepth == 1);
+    assert("-abc".parseMatchStringDepth == 1);
+    assert("abc.foo".parseMatchStringDepth == 2);
+    assert("-abc.foo".parseMatchStringDepth == 2);
+}
+
+/**
+ * Parses a 'match string', which is the "include import" components
+ * given on the command line, i.e. "-i=<match_string>,<match_string>,...".
+ * Params:
+ *  matchString = The match string to parse.
+ *  dst = the data structure to save the parsed match string to.
+ *  depth = the depth of the match string previously retrieved from $(D parseMatchStringDepth).
+ */
+private void parseMatchString(const(char)* matchString, MatcherNode* dst, ushort depth)
```

Comment:
There are quite a few private symbols with Ddoc comments here. I'm not sure if we want them to show up.

Reply:
I'm not sure if these function will end up here or in another module so I'm not sure whether or not they will remain private or become public.  For now I've removed the second `*` to make then "non-ddoc" comments.

Reply:
That's prefect. It's nice to have private functions documented as well when reading the source code.